mlflow/models/evaluation/base.py

```diff
@@ -1581,8 +1581,9 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
                 targets = data.targets
             else:
                 raise MlflowException(
-                    message="The targets argument is required when data is a Dataset and does not "
-                    "define targets.",
+                    message="The targets parameter must be specified with the provided Dataset "
+                    f"for {model_type} models. For example: "
+                    "`data = mlflow.data.from_pandas(df=X.assign(y=y), targets='y')`",
```

Comment:
I think this is still a bit unclear. Can we say that the targets must be specified by the dataset, rather than using the `targets` argument in `mlflow.evaluate()`?